add `rebuild_cmdstan` to handle Xcode updates

#### Summary:
After updating Xcode there were compilation errors stemming from:

```
error: PCH file uses an older PCH format that is no longer supported
```

To solve it required rebuilding ```cmdstan``` which should be suggested by a message in error handling.

#### Description:
After updating Xcode on my Mac my CmdStanPy models stopped working.

``` 
ERROR:cmdstanpy:file /var/folders/96/_4tl8xsx4tx9rj0ykk82715h0000gn/T/tmp_j9qztrd/tmpq_i7gfcf.stan, exception ERROR
error: PCH file uses an older PCH format that is no longer supported
1 error generated.
make: *** [/var/folders/96/_4tl8xsx4tx9rj0ykk82715h0000gn/T/tmp_j9qztrd/tmpq_i7gfcf] Error 1 
ERROR:cmdstanpy:model compilation failed
```

After some searching I found this [link](https://bleepcoder.com/cmdstanr/728633795/error-pch-file-uses-an-older-pch-format-that-is-no-longer) discussing ```cmdstanr```.
Solution proposed there was run ```rebuild_cmdstan()``` function. 
Because there is no such function in ```cmdstanpy``` I ran ```install_cmdstan --overwrite``` which solved the issue.

My suggestion is to improve the error handling here to provide a message along the lines:

``` If your c compiler was recently modified try rebuilding cmdstan``` 


#### Additional Information:

Mac mini (M1, 2020), MacOS BigSur 11.3, Xcode 12.5 

#### Current Version:

Python 3.7.10 64-bit
CmdStan 2.26.1
CmdStanPy 0.9.75

--- a/cmdstanpy/model.py
+++ b/cmdstanpy/model.py
@@ -336,6 +336,21 @@ class CmdStanModel:
                         get_logger().error(
                             'file %s, exception %s', stan_file, str(e)
                         )
+                        if 'PCH file' in str(e):
+                            get_logger().warning(
+                                "%s, %s",
+                                "CmdStan's precompiled header (PCH) files ",
+                                "may need to be rebuilt.",
+                            )
+                            get_logger().warning(
+                                "%s %s",
+                                "If your model failed to compile please run ",
+                                "install_cmdstan(overwrite=True).",
+                            )
+                            get_logger().warning(
+                                "If the issue persists please open a bug report"
+                            )
+
                         compilation_failed = True
 
                 if not compilation_failed:

